source/extensions/filters/common/ext_authz/check_request_utils.cc

```diff
@@ -134,25 +167,32 @@ void CheckRequestUtils::setHttpRequest(
 
   // Fill in the headers.
   auto* mutable_headers = httpreq.mutable_headers();
+  auto* mutable_header_map = httpreq.mutable_header_map();
 
-  headers.iterate([request_header_matchers, mutable_headers](const Envoy::Http::HeaderEntry& e) {
+  headers.iterate([headers_as_bytes, request_header_matchers, mutable_headers,
+                   mutable_header_map](const Envoy::Http::HeaderEntry& e) {
     // Skip any client EnvoyAuthPartialBody header, which could interfere with internal use.
-    if (e.key().getStringView() != Headers::get().EnvoyAuthPartialBody.get()) {
+    if (e.key().getStringView() == Headers::get().EnvoyAuthPartialBody.get()) {
+      return Envoy::Http::HeaderMap::Iterate::Continue;
+    }
+
+    const std::string key(e.key().getStringView());
+    if (request_header_matchers != nullptr && !request_header_matchers->matches(key)) {
+      return Envoy::Http::HeaderMap::Iterate::Continue;
+    }
+
+    if (headers_as_bytes) {
+      addOrAppendHeader(*mutable_header_map, key, e.value().getStringView());
+    } else {
       const std::string sanitized_value =
           MessageUtil::sanitizeUtf8String(e.value().getStringView());
 
-      const std::string key(e.key().getStringView());
-
-      if (request_header_matchers == nullptr || request_header_matchers->matches(key)) {
-        if (mutable_headers->find(key) == mutable_headers->end()) {
-          (*mutable_headers)[key] = sanitized_value;
-        } else {
-          // Merge duplicate headers.
-          (*mutable_headers)[key].append(",").append(sanitized_value);
-        }
+      if (mutable_headers->find(key) == mutable_headers->end()) {
+        (*mutable_headers)[key] = sanitized_value;
+      } else {
+        (*mutable_headers)[key].append(",").append(sanitized_value); // Merge duplicate headers.
```

Comment:
nit: Put the comments on top of the line instead of the back to comply with Envoy coding style.